src/sale/Search/BulkAction.php

```diff
@@ -20,17 +21,26 @@ class BulkAction
      * @var SaleRepositoryInterface
      */
     private $repo;
+    /**
+     * @var HistoryAndFutureSaleRepository
+     */
+    private HistoryAndFutureSaleRepository $historyRepo;
 
-    public function __construct(SaleRepositoryInterface $repo)
+    public function __construct(SaleRepositoryInterface $repo, HistoryAndFutureSaleRepository $saleRepository)
     {
         $this->repo = $repo;
+        $this->historyRepo = $saleRepository;
     }
 
     public function __invoke(Command $command): ArrayCollection
     {
-        $res = $this->repo->findAll(
-            AuthRule::currentUser()->applyToSpecification($command->getSpecification())
-        );
+        $specification = AuthRule::currentUser()->applyToSpecification($command->getSpecification());
+
+        if (in_array('history', $command->include, true)) {
```

Comment:
$res = in_array('history', $command->include, true) 
            ? $this->historyRepo->findAll($specification)
            : $this->repo->findAll($specification);